src/sonic-pit/pit-sysdiag/README.md

```diff
@@ -0,0 +1,1029 @@
+# `sysdiag` 设计说明
+`sysdiag`是一套交换机软硬件测试工具集，目的是对交换机进行健康检查，生成测试报告，并支持环境类测试、研发测试和生产测试等。
+
+`sysdiag`的设计理念是：
+
+ * 测试项通用化，不依赖某硬件平台或特殊场景；
+ * 测试项的具体测试流程需大家认可；
+ * 测试项测试过程中需要输入的内容均采用读配置文件的形式；
+ * 测试项中涉及平台数据获取的调用，均可通过`plugins`下的模块；
+
+由于每个平台硬件设计的差异，可以通过平台层的配置文件做隔离。
+
+### 1 整体架构
+
+`sysdiag`整体架构如下图。`sysdiag`包括两部分`test_cases`和`tools`，其中`test_cases`是测试项集合，`tools`是平台工具。
+
+```json
+      ┌───────────────────────────────────────────────────────────┐
+      │                        sysdiag                            │
+      │  ┌────────────────┬───────────────┐      ┌─────────────┐  │
+      │  │ test_caseA     │   fantool     │      │             │  │
+      │  │ test_caseB     │   psutool     │      │  traffic    │  │
+      │  │ test_caseZ     │   ...         │      │             │  │
+      │  └────────────────┴───────────────┘      └─────────────┘  │
+      │           ▲               ▲                ▲              │
+      │           │               │                │              │
+      └───────────┼───────────────┼────────────────┼──────────────┘
+                  │               │                │
+         ┌────────┴────────┬──────┴──────────┬─────┴───────────┐
+middle   │                 │                 │                 │
+abstract │   plugins       │ common_tools    │     bcmcmd      │
+layer    │                 │                 │     ctc_shell   │
+         └─────────────────┴─────────────────┴─────────────────┘
+
+
+         ┌─────────────────────────────────┐ ┌─────────────────┐
+         │                                 │ │                 │
+kernel   │           hardware drivers      │ │     asic        │
+space    │                                 │ │                 │
+         └─────────────────────────────────┘ └─────────────────┘
+```
+
+### 2 `plugins`中间抽象层接口设计
+`plugins`是`SONiC`下的平台抽象层，`sysdiag`可以利用并扩充此模块提供的接口作为平台数据来源。
```

Comment:
I would suggest using sonic platform API instead of platform plugins, sonic community is deprecating the sonic platform plugins.

Reply:
Platform API will be in next stage, hopefully 202305 release.